downloadDirectory/src/main/java/org/reactome/release/downloadDirectory/Biopax.java

```diff
@@ -0,0 +1,173 @@
+  static ApplicationContext ctx;
+  static boolean autofix = false;
+  static int maxErrors = 0;
+  static String profile = "notstrict";
+  static String outFormat = "xml";
+  static Validator validator;
+
+  static {
+   logger.info("Running BioPAX");
+    // Biopax validation requires loading of bio ontologies, which takes a few minutes
+    ctx = new ClassPathXmlApplicationContext(new String[]{
+      "META-INF/spring/appContext-validator.xml",
+      "META-INF/spring/appContext-loadTimeWeaving.xml"
+    });
+    validator = (Validator) ctx.getBean("validator");
+  }
+  
+  public static void execute(String username, String password, String host, String port, String database, String releaseNumber) throws Exception {
+	  	releaseNumber = releaseNumber.replace("/", "");
+	  	String biopaxDir = releaseNumber + "_biopax";
+        File biopaxDirFile = new File(biopaxDir);
+        if (!biopaxDirFile.exists()) {
+        	biopaxDirFile.mkdir();
+        }
+        // First, generate owl files. This particular step requires a local maven installation of the PathwayExchange jar
+		for (int i = 0; i < 2; i++) {
+			int biopaxLevel = i + 2;
+			if (i == 0) {
+				logger.info("Generating BioPAX2 Species owl Files...");
+				SpeciesAllPathwaysConverter converter = new SpeciesAllPathwaysConverter();
+				converter.doDump(new String[]{host, database, username, password, port, biopaxDir});
+			} else {
+				logger.info("Generating BioPAX3 Species owl Files...");
+				SpeciesAllPathwaysLevel3Converter level3converter = new SpeciesAllPathwaysLevel3Converter();
+				level3converter.doDump(new String[]{host, database, username, password, port, biopaxDir});
+			}
+			
+			// Rename files, replacing whitespace with underscores
+		    Files.newDirectoryStream(Paths.get(biopaxDir),
+		    	      path -> path.toString().endsWith(".owl"))
+		    	      .forEach(f -> {
+		    	          String spath = f.toFile().getPath();
+		    	          spath = spath.replaceAll(" +", "_");
+		    	          File formattedFile = new File(spath);
+		    	          f.toFile().renameTo(formattedFile);
+		    	      });
+		    // Validate each owl file in the biopax output directory produced by PathwaysConverter
+		    logger.info("Validating owl files...");
+			runBiopaxValidation(biopaxDir);
+
+			// Compress all Biopax and validation files into individual zip files
+			logger.info("Zipping BioPAX" + biopaxLevel + " files...");
+			FileOutputStream biopaxOutputStream;
+			FileOutputStream validatorOutputStream;
+			if (i == 0) {
+				biopaxOutputStream = new FileOutputStream("biopax2.zip");
+				validatorOutputStream = new FileOutputStream("biopax2_validator.zip");
+			} else {
+				biopaxOutputStream = new FileOutputStream("biopax.zip");
+				validatorOutputStream = new FileOutputStream("biopax_validator.zip");
+			}
+			ZipOutputStream biopaxZipStream = new ZipOutputStream(biopaxOutputStream);
+			ZipOutputStream validatorZipStream = new ZipOutputStream(validatorOutputStream);
```

Comment:
It might be more readable to generate the ZipOutputStream in one call doing something like this:
```
ZipOutputStream biopaxZipStream = getBiopaxZipStream(biopaxLevel);
ZipOutputStream validatorZipStream = getValidatorZipStream(biopaxLevel);

private static ZipOutputStream getBiopaxZipStream(int biopaxLevel) throws FileNotFoundException {
	return getZipOutputStream(biopaxLevel == 2 ? "biopax2_validator.zip" : "biopax_validator.zip");
}

private static ZipOutputStream getValidatorZipStream(int biopaxLevel) throws FileNotFoundException {
	return getZipOutputStream(biopaxLevel == 2 ? "biopax2.zip" : "biopax.zip");
}

private static ZipOutputStream getZipOutputStream(String fileName) throws FileNotFoundException {
	return new ZipOutputStream(new FileOutputStream(fileName));
}
```

Reply:
Nice and succinct; I'll get that done